include/mfmg/dealii/amge_host.templates.hpp

```diff
@@ -160,9 +168,11 @@ AMGe_host<dim, MeshEvaluator, VectorType>::compute_local_eigenvectors(
       _eigensolver_params.get<std::string>("type", "lanczos");
   if (eigensolver_type == "lanczos")
   {
+    dealii::Vector<double> initial_vector(n_dofs_agglomerate);
```

Comment:
Move these lines out of if-else branches, all eigensolvers need them.

Reply:
Well we support only one eigensolver...

Reply:
Fair point for matrix-free, but with matrices we still do the same thing. I would just always have initial guess out of if-else, and this way both versions would look the same.